### PR TITLE
MES-4633: initialise state management a mod1

### PR DESCRIPTION
**Pull request by szabikr**
## Description

Initialising the store modules for category AMod1 journey.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/modules/tests/tests.cat-a-mod1.reducer.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  return combineReducers(
+    {
+      version: schemaVersionReducer,
+      category: categoryReducer,
+      activityCode: activityCodeReducer,
+      journalData: journalDataCatAMod1Reducer,
+      preTestDeclarations: preTestDeclarationsReducer,
+      accompaniment: accompanimentReducer,
+      vehicleDetails: vehicleDetailsCatAMod1Reducer,
+      testData: testDataCatBEReducer,
```
Comment (Cullima) [inline src/modules/tests/tests.cat-a-mod1.reducer.ts]: BE reducer?

File: src/pages/communication-page/cat-a-mod1/communication.cat-a-mod1.page.ts [hunk 2]
```diff
@@ -10,7 +10,7 @@ import { getCurrentTest, getJournalData } from '../../../modules/tests/tests.sel
 import { getTests } from '../../../modules/tests/tests.reducer';
 
 // TODO - PREP-AMOD1 - implement category specific reducer
```
Comment (Cullima) [inline src/pages/communication-page/cat-a-mod1/communication.cat-a-mod1.page.ts]: Can the TODO be removed with this change?

File: src/modules/tests/test-data/cat-a-mod1/_tests_/test-data.cat-be.selector.spec.ts [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import { TestData } from '@dvsa/mes-test-schema/categories/AM1';
```
Comment (MarkPatt89) [inline src/modules/tests/test-data/cat-a-mod1/_tests_/test-data.cat-be.selector.spec.ts]: filename references `cat-be`

**Review by Cullima — CHANGES_REQUESTED** [review]
looks good! a couple of small questions

File: src/modules/tests/test-data/cat-a-mod1/test-data.cat-a-mod1.reducer.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import { seriousFaultsReducer } from '../common/serious-faults/serious-faults.reducer';
+
+export const initialState: TestData = {
+  dangerousFaults: {},
+  drivingFaults: {},
+  seriousFaults: {},
+  ETA: {},
+};
+
+export function testDataCatBEReducer(
```
Comment (MarkPatt89) [inline src/modules/tests/test-data/cat-a-mod1/test-data.cat-a-mod1.reducer.ts]: function name references `catBE`

File: src/modules/tests/tests.cat-a-mod1.reducer.ts (hunk 1 above)
Comment (MarkPatt89) [inline src/modules/tests/tests.cat-a-mod1.reducer.ts]: `testDataCatBEReducer` referenced here

**Review by MarkPatt89 — COMMENTED** [review]
A couple of things that need changing with references to `catBE` still in place.

File: src/modules/tests/tests.cat-a-mod1.reducer.ts (hunk 1 above)
Comment (szabikr) [reply src/modules/tests/tests.cat-a-mod1.reducer.ts]: Built the reducer but haven't changed the reference here - pushing up a change

File: src/pages/communication-page/cat-a-mod1/communication.cat-a-mod1.page.ts (hunk 2 above)
Comment (szabikr) [reply src/pages/communication-page/cat-a-mod1/communication.cat-a-mod1.page.ts]: Removing TODO

File: src/modules/tests/test-data/cat-a-mod1/_tests_/test-data.cat-be.selector.spec.ts (hunk 3 above)
Comment (szabikr) [reply src/modules/tests/test-data/cat-a-mod1/_tests_/test-data.cat-be.selector.spec.ts]: changing it to `a-mod1`

File: src/modules/tests/test-data/cat-a-mod1/test-data.cat-a-mod1.reducer.ts (hunk 4 above)
Comment (szabikr) [reply src/modules/tests/test-data/cat-a-mod1/test-data.cat-a-mod1.reducer.ts]: changing it

File: src/modules/tests/tests.cat-a-mod1.reducer.ts (hunk 1 above)
Comment (szabikr) [reply src/modules/tests/tests.cat-a-mod1.reducer.ts]: changing it

**Review by MarkPatt89 — APPROVED** [review]
Looks good.

File: src/pages/debrief/cat-a-mod2/debrief.cat-a-mod2.page.ts [hunk 5]
```diff
@@ -35,7 +35,7 @@ import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/
 import { FaultSummaryProvider } from '../../../providers/fault-summary/fault-summary';
 
 // TODO - PREP-AMOD2 - Implement category specific reducer
```
Comment (Cullima) [inline src/pages/debrief/cat-a-mod2/debrief.cat-a-mod2.page.ts]: remove TODO?

File: src/modules/tests/tests.effects.ts [hunk 6]
```diff
@@ -164,10 +164,6 @@ export class TestsEffects {
 
       // TODO: PREP-AMOD1 Remove bypass logic for enabling a-mod1
```
Comment (Cullima) [inline src/modules/tests/tests.effects.ts]: remove TODO?

File: src/pages/communication-page/cat-a-mod2/communication.cat-a-mod2.page.ts [hunk 7]
```diff
@@ -10,7 +10,7 @@ import { getCurrentTest, getJournalData } from '../../../modules/tests/tests.sel
 import { getTests } from '../../../modules/tests/tests.reducer';
 
 // TODO - PREP-AMOD2 implement category specific reducer
```
Comment (Cullima) [inline src/pages/communication-page/cat-a-mod2/communication.cat-a-mod2.page.ts]: remove?

File: src/pages/debrief/cat-a-mod1/debrief.cat-a-mod1.page.ts [hunk 8]
```diff
@@ -34,7 +34,7 @@ import { TestCategory } from '@dvsa/mes-test-schema/category-definitions/common/
 import { FaultSummaryProvider } from '../../../providers/fault-summary/fault-summary';
 
 // TODO - PREP-AMOD1: Use cat a mod1 reducer
```
Comment (Cullima) [inline src/pages/debrief/cat-a-mod1/debrief.cat-a-mod1.page.ts]: remove?

**Review by Cullima — APPROVED** [review]

File: src/modules/tests/tests.effects.ts (hunk 6 above)
Comment (szabikr) [reply src/modules/tests/tests.effects.ts]: Changing this to remove a-mod2

File: src/pages/debrief/cat-a-mod2/debrief.cat-a-mod2.page.ts (hunk 5 above)
Comment (szabikr) [reply src/pages/debrief/cat-a-mod2/debrief.cat-a-mod2.page.ts]: removing it

File: src/pages/communication-page/cat-a-mod2/communication.cat-a-mod2.page.ts (hunk 7 above)
Comment (szabikr) [reply src/pages/communication-page/cat-a-mod2/communication.cat-a-mod2.page.ts]: removed

File: src/pages/debrief/cat-a-mod1/debrief.cat-a-mod1.page.ts (hunk 8 above)
Comment (szabikr) [reply src/pages/debrief/cat-a-mod1/debrief.cat-a-mod1.page.ts]: removed

**Pull request merged**